src/Symfony/Component/Routing/Tests/RouteCompilerTest.php

```diff
@@ -36,15 +36,15 @@ public function provideCompileData()
             array(
                 'Static route',
                 array('/foo'),
-                '/foo', '#^/foo$#s', array(), array(
+                '/foo', '#^/foo$#us', array(), array(
```

Comment:
The regexp generate by RouteCompiler will always have the `u` flag. All parameters etc. are passed as UTF-8 internally. If the router is configured to use a different encoding, the URL is translated from that encoding to UTF-8 early in UrlMatcher, and vice versa in UrlGenerator.

UrlGeneratorTest has some tests with different charsets. I guess I need to make some in UrlMatcherTest also.